AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs

```diff
@@ -188,32 +197,94 @@ public void EditUserPostReturnsSameViewAndViewModelWhenModelStateIsInvalid()
         }
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task EditUserPostInvokesUpdateUserWithCorrectUserWhenUsersAssociatedSkillsAreNotNullAndThereIsAtLeastOneAssociatedSkillForTheUser()
         {
-            //delete this line when starting work on this unit test
-            await TaskFromResultZero;
+            var mediator = new Mock<IMediator>();
+            var logger = new Mock<ILogger<SiteController>>();
+            EditUserModel model = new EditUserModel()
+            {
+                UserId = It.IsAny<string>(),
+                AssociatedSkills = new List<UserSkill>() { new UserSkill() { SkillId = 1, Skill = new Skill() { Id = 1 } } }
+            };
+            var user = new ApplicationUser()
+            {
+                Id = model.UserId,
+                AssociatedSkills = new List<UserSkill>() { new UserSkill() { SkillId = 2, Skill = new Skill() { Id = 2 } } }
+            };
+            mediator.Setup(x => x.Send(It.Is<UserByUserIdQuery>(q => q.UserId == model.UserId)))
+                .Returns(user);
+
+            var controller = new SiteController(null, logger.Object, mediator.Object);
+
+            await controller.EditUser(model);
+            mediator.Verify(m => m.SendAsync(It.Is<UpdateUser>(q => q.User.AssociatedSkills[0] == model.AssociatedSkills[0])), Times.Once);
         }
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task EditUserPostInvokesAddClaimAsyncWhenModelsIsOrganizationAdminIsTrue()
         {
-            //delete this line when starting work on this unit test
-            await TaskFromResultZero;
+            var mediator = new Mock<IMediator>();
+            var userManager = CreateApplicationUserMock();
+            var logger = new Mock<ILogger<SiteController>>();
+            EditUserModel model = new EditUserModel()
+            {
+                IsOrganizationAdmin = true,
+                UserId = It.IsAny<string>()
+                
+            };
+            var user = new ApplicationUser()
+            {
+                Id = model.UserId,
+                Email = "test@testy.com"
+            };
+
+            mediator.Setup(x => x.Send(It.Is<UserByUserIdQuery>(q => q.UserId == model.UserId)))
+                .Returns(user);
+            userManager.Setup(x => x.AddClaimAsync(It.IsAny<ApplicationUser>(), It.IsAny<Claim>()))
+                .Returns(() => Task.FromResult(IdentityResult.Success));
+
+            var controller = new SiteController(userManager.Object, logger.Object, mediator.Object);
+            controller.SetDefaultHttpContext();
+            controller.Url = CreateUrlHelperMockObject().Object;
+            await controller.EditUser(model);
+
+            userManager.Verify(x => x.AddClaimAsync(It.IsAny<ApplicationUser>(), It.Is<Claim>(c => c.Value == "OrgAdmin") ), Times.Once);
```

Comment:
we should assert that .AddClaimAsync was called with a specific user and Claim. Here you're saying `It.IsAny<ApplicationUser>`... You control the user that `UserByUserIdQuery` is returning earlier in your test method, and you can copy the claim code from the action method into the unit test... so something like this:
```
var organizationAdminClaim = new Claim(AllReady.Security.ClaimTypes.UserType, "OrgAdmin");
userManager.Verify(x => x.AddClaimAsync(user, organizationAdminClaim), Times.Once);
```

I haven't run this, so I don't know if it will work, but give it a shot.